homeworks/lesia.turua_lesiaturuta/PokemonProject/scripts/pokemon.js

```diff
@@ -309,58 +66,138 @@ const renderCards = (cardContainer, pokemonsArray) => {
     return true;
 };
 
-const cleanElement = (el) => {
-    while (el.firstChild) {
-        el.removeChild(el.firstChild);
+const renderSelect = (name) => {
+    cleanElement(searchEl);
+    const optionFirstEl = document.createElement('option');
+    if (name === '') {
+        optionFirstEl.value = 'Select type: ';
+        optionFirstEl.text = 'Select type: ';
+        optionFirstEl.selected = true;
+        optionFirstEl.disabled = true;
+    } else {
+        optionFirstEl.value = '';
+        optionFirstEl.text = 'All types';
+    }
+
+    searchEl.add(optionFirstEl);
+    typesPokemons.forEach((type) => {
+        const optionEl = document.createElement('option');
+        optionEl.value = type;
+        if (type === name) {
+            optionEl.selected = true;
+        }
+        optionEl.text = type;
+        searchEl.add(optionEl);
+    });
+};
+
+const sortPokemonsUp = (arrayPokemons) => arrayPokemons.sort((a, b) => a.height - b.height);
+
+const sortPokemonsDown = (arrayPokemons) => arrayPokemons.sort((a, b) => b.height - a.height);
+
+const updateTypes = (array) => {
+    const types = [];
+    array.forEach((pokemon) => pokemon.type.forEach((type) => {
+        if (!types.includes(type)) types.push(type);
+    }));
+    return types;
+};
+
+const getPokemonsByType = (type, arrayPokemons) => arrayPokemons
+    .filter((pokemon) => pokemon.type
+        .includes(type));
+const getPokemonsByValet = (text, arrayPokemons) => arrayPokemons
+    .filter((pokemon) => pokemon.name
+        .toLocaleLowerCase()
+        .includes(text.toLocaleLowerCase()));
+
+const getPokemons = async (urlPokemons) => {
+    try {
+        const response = await fetch(urlPokemons);
+        return await response.json();
+    } catch (e) {
+        alert('Error url, please update this page');
+        console.error(e);
+        return [];
     }
 };
 
-const filterPokemons = (checkBig, checkSmall) => {
-    let newPokemons = pokemons;
-    if (checkBig && checkSmall) {
-        newPokemons = pokemons.filter((pokemon) => pokemon.weight < 100 && pokemon.weight > 50);
-    } else if (checkBig && !checkSmall) {
-        newPokemons = pokemons.filter((pokemon) => pokemon.weight > 100);
-    } else if (!checkBig && checkSmall) {
-        newPokemons = pokemons.filter((pokemon) => pokemon.weight < 50);
+const getFilterPokemons = () => {
+    let newPokemons = [];
+    if (searchValue && typeValue) {
```

Comment:
it is so easy to lost in all these if-else
could you make some auxiliary functions, to make this block simpler?
```
newPokemons = [... pokemons]

newPokemons = getPokemonsByName(newPokemon)  // logic of empty search is inside

newPokemons = getPokemonsByType(newPokemons)  // logic of not selected any type inside

newPokemons = getPokemonsByWeight(newPokemons) // inside you get checked size checkboxes and filter pokemons by them

newPokemons = sortByHeight(newPokemons) 
```